### PR TITLE
chore(ava): Rewrite reducer tests in ava and remove mocha

### PR DESCRIPTION
**Pull request by JaKXz**
Removes un-needed coverage plugin, unused eslintignore, and add concurrently for lint tasks since
eslint is slow.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,15 +12,25 @@
     "build": "cross-env NODE_ENV=production webpack -p",
     "clean": "rimraf dist/",
     "dev": "cross-env NODE_ENV=development webpack-dev-server -d --inline --hot --progress --no-info",
-    "lint": "npm run lint-js && npm run lint-css",
+    "lint": "concurrently 'npm run lint-js' 'npm run lint-css'",
```
Comment (RSNara) [inline package.json:15]: Love this concurrently module. Wasted so much of my life in the past dealing with this nonsense.

**Comment by JaKXz** [comment]
@rsnara for the sake of small-ish PRs, this is a start :) other component and reducer tests can follow asynchronously

File: src/reducers/session.test.js [hunk 2]
```diff
@@ -11,50 +11,38 @@ import {
...
-    it('should save the username', () => {
-      state = fireAction(sessionReducer, state, LOGOUT_USER);
-
-      assert(!state.get('isLoading'));
-      assert(!state.get('hasError'));
-      assert(state.get('token') === null);
-    });
-  });
+test.beforeEach(t => {
+  t.context.state = sessionReducer(undefined, {});
```
Comment (RSNara) [inline src/reducers/session.test.js:15]: So `t.context` is shared between all tests?

File: package.json [hunk 3]
```diff
@@ -122,18 +131,21 @@
...
+    "reselect"
   ],
   "repository": {
     "type": "git",
-    "url": "git://github.com/rangle/react-redux-starter.git"
+    "url": "git://github.com/rsnara/reddit-client.git"
   },
   "bugs": {
-    "url": "https://github.com/rangle/rangle-starter/issues"
+    "url": "https://github.com/rsnara/reddit-client/issues"
```
Comment (RSNara) [inline package.json:142]: Sigh... `rangle-starter` actually asks for the github username. I wonder why they don't modify the package.json file appropriately. Maybe we should do a pull request.

**Comment by RSNara** [comment]
Looks good!

File: src/reducers/session.test.js (hunk 2 above)
Comment (JaKXz) [inline src/reducers/session.test.js:15]: Nope, `t.context` is shared between the `beforeEach` function and then there's one for each test that follows. https://github.com/avajs/ava#before--after-hooks

**Pull request merged**